packages/harmony/src/components/button/FilterButton/FilterButton.tsx

```diff
@@ -0,0 +1,218 @@
+import {
+  forwardRef,
+  ForwardedRef,
+  RefObject,
+  useRef,
+  useState,
+  useCallback
+} from 'react'
+
+import { CSSObject, useTheme } from '@emotion/react'
+
+import { BaseButton } from 'components/button/BaseButton/BaseButton'
+import { Box, Flex, Paper, Popup } from 'components/layout'
+import { IconCaretDown, IconCloseAlt } from 'icons'
+
+import {
+  FilterButtonOption,
+  FilterButtonProps,
+  FilterButtonSize,
+  FilterButtonType
+} from '../types'
+
+export const FilterButton = forwardRef<HTMLButtonElement, FilterButtonProps>(
+  function FilterButton(
+    props: FilterButtonProps,
+    ref: ForwardedRef<HTMLButtonElement>
```

Comment:
do you need to add props and ref types here? i think with the `forwardRef<ref, props>` pattern it should cover for you

Reply:
IDE was saying this was invalid and also some of the other places we don't do this, though harmony seems inconsistent. Maybe I have some config incorrect b/c I do get a lot of type errors in harmony in general w/ imports

Reply:
dont stress, we can take a look at some point